substrate/frame/bags-list/src/lib.rs

```diff
@@ -460,9 +481,14 @@ pub mod pallet {
 					Self::iter()
 				},
 			};
-			let accounts: Vec<_> = iter.take((rebag_budget + 1) as usize).collect();
 
-			// Safe split: if we reached (or passed) the tail of the list, we don’t want to panic.
+			// Chain PendingRebag accounts with regular ListNodes.
+			// PendingRebag comes first for priority processing
+			let combined_iter = PendingRebag::<T, I>::iter_keys().chain(regular_iter);
```

Comment:
```suggestion
			let combined_iter = PendingRebag::<T, I>::iter_keys().drain().chain(regular_iter);
```

first glance review: I think this API can simplify the code. It will instruct the iterator to also clean each item as it is being processed.

Reply:
We don't want to remove items from storage before we know we have budget for it and drain() would do that, right?
I think the current approach is actually better: we remove only after confirmed success, we don't modify storage until we know we have budget for it , failed items remain there for retry